### ISSUE TITLE
andalso, orelse

### ISSUE THREAD

**Issue by michallepicki** [gfngfn/Sesterl#43]
I find myself needing to write more complicated boolean expressions. Ideally they would also be "lazy" so wouldn't compute the right expression if left expression already gives the result.

I searched for them and only found [this commit](https://github.com/gfngfn/Sesterl/commit/5da8eef34e24dc3b273b6194e88d5ceb5b873cda).

Are such operators (or "functions" with special semantics) planned?

**Comment by michallepicki** [comment]
I guess `andalso` is `if left then right else false`
and `orelse` is `if left then true else right

**Comment by gfngfn** [comment]
Yes, I understand the need for short circuit logical operators, but I haven’t decided how to provide them; they can be implemented as:

1. special built-in operators (like Erlang’s `andalso` and `orelse`), or
2. macros (after introducing some type-safe mechanism for writing macros).

The first one is easy to realize but somewhat ad-hoc, while the second one seems theoretically elegant and general but a bit tough to achieve.
